Updating existing blog article to 2025 trends (#12932)
pulumi/docs: content/blog/future-cloud-infrastructure-10-trends-shaping-2024-and-beyond/index.md

---
title: "Future of the Cloud: 10 Trends Shaping 2025 and Beyond"

# The date represents the post's publish date, and by default corresponds with
# the date and time this file was generated. Dates are used for display and
# ordering purposes only; they have no effect on whether or when a post is
# published. To influence the ordering of posts published on the same date, use
# the time portion of the date value; posts are sorted in descending order by
# date/time.

> date: 2024-10-03T07:56:40Z

Author: ```suggestion
date: 2024-11-06T07:56:40Z
```

# published. To influence the ordering of posts published on the same date, use
# the time portion of the date value; posts are sorted in descending order by
# date/time.

> date: 2025-10-09T07:56:40Z

Author: ```suggestion
date: 2024-10-03T07:56:40Z
```

# The meta_image appears in social-media previews and on the blog home page. A
# placeholder image representing the recommended format, dimensions and aspect
# ratio has been provided for you.

> meta_image: cloud-computing-forecast.png

Author: ```suggestion
meta_image: "cloud-computing-forecast.png.png"
```

[Infrastructure as Code](https://www.pulumi.com/what-is/what-is-infrastructure-as-code/) is maturing beyond taming the complexity of the cloud:

- Facilitating the adoption and configuration standardization of multi-cloud and hybrid strategies
- Better integration with multiple cloud providers, like [AWS](https://www.pulumi.com/docs/iac/clouds/aws/), [Azure](https://www.pulumi.com/docs/iac/clouds/azure/), and [Google Cloud](https://www.pulumi.com/docs/iac/clouds/gcp/), data stores, and third-party services like [Cockroach Labs db](https://www.pulumi.com/registry/packages/cockroach/), [Confluent cloud](https://www.pulumi.com/registry/packages/confluentcloud/), [Kafka](https://www.pulumi.com/registry/packages/kafka/), and more
- Deeper validation on parameters that people are passing in, checking all critical components, ensuring they are configured correctly before deployment
- More [efficient resource management](https://www.pulumi.com/docs/pulumi-cloud/insights/)

> - Robust security and [Policy as Code]([https://www.pulumi.com/docs/using-pulumi/crossguard/core-concepts/](https://www.pulumi.com/docs/iac/packages-and-automation/crossguard/core-concepts/)) to enforce security practices, guardrails, compliance, cost policies, and more

Author: ```suggestion
```

As a result, [IDC](https://www.netapp.com/media/93564-operationalize-fin-ops-for-continuous-cloud-and-cost-efficiency.pdf) predicts that complexities and IT budget pressures will drive **70% of Global 1000 companies to [increase FinOps maturity](https://www.pulumi.com/blog/finops-with-pulumi/)** with granular chargebacks, benchmarking, and multiple-cloud optimization by 2024.

## 10. Enterprises Will allow the Use of AI Code Assistants

Developers worldwide have explored or are currently using AI-powered coding assistants. However, many enterprises have shown resistance to allowing them to be part of their AI tools in software development. Still, software engineering leaders are beginning to recognize that these coding assistants can enhance team productivity, improve code quality, and maintain a competitive advantage.

> **By 2027, AI will dramatically increase developer velocity by automatically generating code** to meet functional business requirements for 70% of new digital solutions in production (source: [IDC](https://www.digitalnewsasia.com/business/idc-reveals-its-top-predictions-cloud-2023-and-beyond)).

Author: ```suggestion
**By 2027, the use of [AI assistants](https://www.pulumi.com/product/copilot/) will dramatically increase developer velocity** to meet functional business requirements for 70% of new digital solutions in production (source: [IDC](https://www.digitalnewsasia.com/business/idc-reveals-its-top-predictions-cloud-2023-and-beyond)).
```

{{< figure alt="The value of AI code assistants. Credit: Gartner" src="/blog/future-cloud-infrastructure-10-trends-shaping-2024-and-beyond/ai_code_assistants_value.png" caption="The value of AI code assistants. Credit: Gartner" width=100% >}}

> According to [Gartner](https://www.gartner.com/en/newsroom/press-releases/2024-04-11-gartner-says-75-percent-of-enterprise-software-engineers-will-use-ai-code-assistants-by-2028), **by 2028, 75% of enterprise software engineers will use dedicated AI code assistants**, and 63% of organizations are currently piloting, deploying or beginning to use AI code assistants.

Author: ```suggestion
According to [Gartner](https://www.gartner.com/en/newsroom/press-releases/2024-04-11-gartner-says-75-percent-of-enterprise-software-engineers-will-use-ai-code-assistants-by-2028), **by 2028, 75% of enterprise software engineers will use dedicated AI code assistants**, and 63% of organizations are currently piloting, deploying or beginning to [use AI code assistants, just like Pulumi AI](https://www.pulumi.com/ai).
```
